backend/src/main/java/park_su_park/backend/controller/UserController.java

```diff
@@ -41,47 +39,31 @@ public ResponseEntity<ResponseUserDto> createUser(
     public ResponseEntity<ResponseUserDto> readUser(
         @RequestParam(name = "username", required = false) String username,
         @RequestParam(name = "email", required = false) String email) {
-        //username email 동시 조회 불가
+
         if (username != null && email != null) {
             throw new ValidateException("username과 email 중 하나만 입력하세요");
         }
-        //username으로 조회
-        else if (username != null) {
-            Optional<User> byUsername = userRepository.findByUsername(username);
-            if (byUsername.isPresent()) {
-                ResponseUserDto responseUserDto = ResponseUserDto.of(byUsername.get());
-                return ResponseEntity.ok(responseUserDto);
-            } else {
-                throw new NotExistUserException("등록되지 않은 사용자입니다.");
-            }
-            //email 조회
-        } else if (email != null) {
-            Optional<User> byEmail = userRepository.findByEmail(email);
-            if (byEmail.isPresent()) {
-                ResponseUserDto responseUserDto = ResponseUserDto.of(byEmail.get());
-                return ResponseEntity.ok(responseUserDto);
-            } else {
-                throw new NotExistUserException("등록되지 않은 사용자 입니다.");
-            }
-            //파라미터가 없는 경우
-        } else {
-            throw new ValidateException("파라미터를 입력하세요");
-        }
+
+        User user = userService.findUserByUsernameOrEmail(username, email);
+        ResponseUserDto responseUserDto = ResponseUserDto.of(user);
+
+        return ResponseEntity.ok(responseUserDto);
     }
 
+
     //U
-    @PostMapping("/update/{id}")
-    public ResponseEntity<ResponseUserDto> updateUser(@PathVariable(name = "userid") Long id,
+    @PostMapping("/update/{userId}")
+    public ResponseEntity<ResponseUserDto> updateUser(@PathVariable(name = "userid") Long userId,
```

Comment:
```suggestion
    public ResponseEntity<ResponseUserDto> updateUser(@PathVariable(name = "userId") Long userId,
```

오타가 있는 것 같습니다.

그리고 변수명과 파라미터 명이 동일하면 name 안넘겨줘도 되지 않나요?

Reply:
개념이 부족한거 같아서 강의 듣는 중입니다.. 양해부탁드려요 ㅜ